src/analyzer.ts

```diff
@@ -82,102 +90,117 @@ export class Analyzer {
 
   /**
    * Loads, parses and analyzes a document and its transitive dependencies.
-   *
-   * @param {string} url the location of the file to analyze
-   * @return {Promise<DocumentDescriptor>}
    */
-  async analyze(url: string): Promise<DocumentDescriptor> {
-    return this._analyzeResolved(this._resolveUrl(url));
+  async analyze(url: string): Promise<Document|null> {
+    const scannedDocument = await this._analyzeResolved(this._resolveUrl(url));
+    return Document.makeRootDocument(scannedDocument);
   }
 
-  private async _analyzeResolved(resolvedUrl: string):
-      Promise<DocumentDescriptor> {
-    let isExternalUrl = urlLib.parse(resolvedUrl).protocol !== null ||
-        resolvedUrl.startsWith('//');
-    if (isExternalUrl) {
-      return null;
-    }
-    let cachedResult = this._documentDescriptors.get(resolvedUrl);
+  /**
+   * Like `analyze`, but for use cases like editors where the caller believes
+   * the file has changed since the last time it was analyzed, and you may
+   * even have unsaved in-memory contents to use for the given file.
+   */
+  async analyzeChangedFile(url: string, contents?: string):
```

Comment:
removed and added a `contents` parameter to `analyze()` that triggers cache clearing.